adapters/uniqueid_linux.c

```diff
@@ -25,13 +39,38 @@ UNIQUEID_RESULT UniqueId_Generate(char* uid, size_t len)
     }
     else
     {
-        uuid_t uuidVal;
-        uuid_generate(uuidVal);
 
-        /* Codes_SRS_UNIQUEID_07_001: [UniqueId_Generate shall create a unique Id 36 character long string.] */
-        memset(uid, 0, len);
-        uuid_unparse(uuidVal, uid);
-        result = UNIQUEID_OK;
+        // generate version 4 uuid
+        UUID unique_id;
+
+        // get random bytes
+        ssize_t bytes_got = getrandom(&unique_id, sizeof(unique_id), GRND_NONBLOCK);
+        if (bytes_got != sizeof(unique_id)) {
+            LogError("Failed to obtain random numbers from getrandom.");
+            result = UNIQUEID_ERROR;
+        }
+        else
+        {
+            // format as version 4 (random) uuid variant 2
```

Comment:
I would like to see this "unparse" section as a separate function.  It will also need a unit test.
Our unit tests for UniqueId_Generate only check for expected final string length or null input. 
Unit tests for this are found in tests/uniqueid_ut/uniqueid_ut_linux.c